### ISSUE TITLE
The prediction score is -1

### ISSUE THREAD

**Issue by LCorleone** [fizyr/keras-retinanet#622]
Hi, i met a confusing problem and i see  #618 said -1 means the model detects nothing.
```
    for box, score, label in zip(boxes[0], scores[0], labels[0]):
        # scores are sorted so we can break
        print(score)
        if score < 0.7:
            break
```
![image](https://user-images.githubusercontent.com/22718985/43674148-672d0d44-9801-11e8-8bfb-7653c90273b4.png)
I doubt that if the model predicts nothing, what the box means? and why it outputs box coordinates? instead I think it should output nothing.

**Comment by borakrc** [comment]
Boxes that have a corresponding score of -1. are and should be ignored.

Because changing the size of a vector is more complicated than changing its values to -1, this is usually considered best practice.

A side note, as far as I know, if a neuron in a model is supposed to have an output for some input, it must have an output for every input. The output can be 1, 0.5, 0, -1, any real number for different inputs. But never null (type None).

**Comment by borakrc** [comment]
Oh and in this case, the score is actually a probability and should be a floating number between 0 and 1. Value -1 is obviously invalid. In cases where value -1 can be valid, you won't see this practice.

**Comment by LCorleone** [comment]
@borakrc Thank you for your reply.
Maybe i find the reasons.
at
> keras-retinanet\keras_retinanet\layers

the comments
```
    Returns
        A list of [boxes, scores, labels, other[0], other[1], ...].
        boxes is shaped (max_detections, 4) and contains the (x1, y1, x2, y2) of the non-suppressed boxes.
        scores is shaped (max_detections,) and contains the scores of the predicted class.
        labels is shaped (max_detections,) and contains the predicted label.
        other[i] is shaped (max_detections, ...) and contains the filtered other[i] data.
        In case there are less than max_detections detections, the tensors are padded with -1's.
```
I think this solves my problem.

**Comment by hgaiser** [comment]
@borakrc answered this perfectly and indeed, as the comment shows, the detections are passed so that the output dimension is always the same. This allows outputting with batch-size > 1. Otherwise, how are you going to return a result where the first image has 3 objects and the second image 5 objects?

Closing this as it is resolved.

**Comment by LCorleone** [comment]
@hgaiser 
Hi, thank you for your reply. what you mentioned also confuses me, I didn't do anything.
as I mentioned above, if I set max_detections=300，it will outputs 300 scores? indeed, the picture above is when max_detections=300. but it outputs various of socres!
I just want to train a model to detect safety hat and finetune the resnet50 model as usual, adding a new label and prepare the csv file.
do you have any ideas about the issue? thank you!

**Comment by LCorleone** [comment]
@hgaiser 
Oh, another question.
how to measure if the model is trained well? there are no validation set? my train loss is always decreasing and I train 70 epoches now.
